google-cloud-examples/src/main/java/com/google/cloud/examples/storage/objects/BatchSetObjectMetadata.java

```diff
@@ -0,0 +1,51 @@
+package com.google.cloud.examples.storage.objects;
+
+// [START storage_batch_request]
+import com.google.api.gax.paging.Page;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageBatch;
+import com.google.cloud.storage.StorageOptions;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BatchSetObjectMetadata {
+  public static void batchSetObjectMetadata(
+      String projectId, String bucketName, String directoryPrefix) {
+    // The ID of your GCP project
+    // String projectId = "your-project-id";
+
+    // The ID of your GCS bucket
+    // String bucketName = "your-unique-bucket-name";
+
+    // The directory prefix. All objects in the bucket with this prefix will have their metadata
+    // updated
+    // String directoryPrefix = "yourDirectory/";
+
+    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    Map<String, String> newMetadata = new HashMap<>();
+    newMetadata.put("keyToAddOrUpdate", "value");
+    Page<Blob> blobs =
+        storage.list(
+            bucketName,
+            Storage.BlobListOption.prefix(directoryPrefix),
+            Storage.BlobListOption.currentDirectory());
+    StorageBatch batchRequest = storage.batch();
+
+    // Add all blobs with the given prefix to the batch request
+    for (Blob blob : blobs.iterateAll()) {
+      batchRequest.update(blob.toBuilder().setMetadata(newMetadata).build());
```

Comment:
Is there a max number of updates you can put into a batch? If someones has lots of objects I could imagine this possibly resulting in an error.

Reply:
According to the [docs](https://cloud.google.com/storage/docs/batch), "You should not include more than 100 calls in a single batch request. If you need to make more calls than that, use multiple batch requests. The total batch request payload must be less than 10MB."

Should i put a note about that in the comments?

Reply:
I would recommend limiting the batch operation to 100 calls per batch operation / submit combination. 

Might be overkill though?

Reply:
@frankyn Like, in this sample? Or do you mean update the actual batch code?

Reply:
This is more of a question; do you know if the Batch client will handle batch requests correctly if there are more than 100 requests?

Otherwise LGTM.